### PR TITLE
JAMES-2765 Use ES 6 for Quota Search

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: backends-common/elasticsearch-v6/src/test/java/org/apache/james/backends/es/v6/search/ScrollIterableTest.java [hunk 1]
```diff
@@ -50,7 +50,7 @@
 
 public class ScrollIterableTest {
 
-    private static final TimeValue TIMEOUT = new TimeValue(6000);
+    private static final TimeValue TIMEOUT = TimeValue.timeValueMinutes(1);
```
Comment (trantienduchn) [inline backends-common/elasticsearch-v6/src/test/java/org/apache/james/backends/es/v6/search/ScrollIterableTest.java]: I think it should be 6 seconds due to the old code `new TimeValue(6000)`

File: backends-common/elasticsearch-v6/src/main/java/org/apache/james/backends/es/v6/ListenerToFuture.java [hunk 2]
```diff
@@ -19,21 +19,33 @@
...
-    private static final Logger LOGGER = LoggerFactory.getLogger(DeleteByQueryActionListener.class);
+public class ListenerToFuture<T> implements ActionListener<T> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ListenerToFuture.class);
 
-    @Override
-    public void onResponse(BulkByScrollResponse bulkByScrollResponse) {
+    private CompletableFuture<T> future;
 
+    public ListenerToFuture() {
+        this.future = new CompletableFuture<>();
```
Comment (blackheaven) [inline backends-common/elasticsearch-v6/src/main/java/org/apache/james/backends/es/v6/ListenerToFuture.java]: Why do you assign an empty CF?

File: backends-common/elasticsearch-v6/src/test/java/org/apache/james/backends/es/v6/search/ScrollIterableTest.java (hunk 1 above)
Comment (chibenwa) [reply backends-common/elasticsearch-v6/src/test/java/org/apache/james/backends/es/v6/search/ScrollIterableTest.java]: 6 seconds is rather short - I already noticed that infinite scroll was blocking at some point in OpenPaaS Inbox. Corelated?

I'd rather keep the same constants than in other places.

File: backends-common/elasticsearch-v6/src/main/java/org/apache/james/backends/es/v6/ListenerToFuture.java (hunk 2 above)
Comment (chibenwa) [reply backends-common/elasticsearch-v6/src/main/java/org/apache/james/backends/es/v6/ListenerToFuture.java]: To complete it later ^^

#mutability <3

File: backends-common/elasticsearch-v6/src/test/java/org/apache/james/backends/es/v6/search/ScrollIterableTest.java (hunk 1 above)
Comment (chibenwa) [reply backends-common/elasticsearch-v6/src/test/java/org/apache/james/backends/es/v6/search/ScrollIterableTest.java]: Let's do that in another PR - but I belive `6 seconds` is a bit too short.

Comment (chibenwa) [reply backends-common/elasticsearch-v6/src/test/java/org/apache/james/backends/es/v6/search/ScrollIterableTest.java]: Let's do that in another PR - but I belive `6 seconds` is a bit too short.

Comment (chibenwa) [reply backends-common/elasticsearch-v6/src/test/java/org/apache/james/backends/es/v6/search/ScrollIterableTest.java]: Looking at ES 2.x code, the timeoute is `new TimeValue(60000)` hence I'll keep the 1 minute timeout here too.

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**